### ISSUE TITLE
OSX 10.11 LevelDB read failure: Corruption: block checksum mismatch

### ISSUE THREAD

**Issue by Chaintool** [bitcoin/bitcoin#7479]
2016-02-08 15:16:23 Bitcoin version v0.11.2 (2015-11-10 14:46:18 +0100)
2016-02-08 15:16:23 Using OpenSSL version OpenSSL 1.0.1k 8 Jan 2015
2016-02-08 15:16:23 Using BerkeleyDB version Berkeley DB 4.8.30: (April  9, 2010)
2016-02-08 15:16:23 Default data directory /Users/Gary/Library/Application Support/Bitcoin
2016-02-08 15:16:23 Using data directory /Volumes/BLOCKCHAIN/Bitcoin
2016-02-08 15:16:23 Using config file /Volumes/BLOCKCHAIN/Bitcoin/bitcoin.conf
2016-02-08 15:16:23 Using at most 125 connections (2560 file descriptors available)
2016-02-08 15:16:23 Using 5 threads for script verification
2016-02-08 15:16:23 Using wallet wallet.dat
2016-02-08 15:16:23 scheduler thread start
2016-02-08 15:16:23 init message: Verifying wallet...
2016-02-08 15:16:23 CDBEnv::Open: LogDir=/Volumes/BLOCKCHAIN/Bitcoin/database ErrorFile=/Volumes/BLOCKCHAIN/Bitcoin/db.log
2016-02-08 15:16:23 Bound to [::]:8333
2016-02-08 15:16:23 Bound to 0.0.0.0:8333
2016-02-08 15:16:23 Cache configuration:
2016-02-08 15:16:23 * Using 2.0MiB for block index database
2016-02-08 15:16:23 * Using 45.0MiB for chain state database
2016-02-08 15:16:23 * Using 103.0MiB for in-memory UTXO set
2016-02-08 15:16:23 init message: Loading block index...
2016-02-08 15:16:23 Opening LevelDB in /Volumes/BLOCKCHAIN/Bitcoin/blocks/index
2016-02-08 15:16:23 Opened LevelDB successfully
2016-02-08 15:16:23 Opening LevelDB in /Volumes/BLOCKCHAIN/Bitcoin/chainstate
2016-02-08 15:16:23 Opened LevelDB successfully
2016-02-08 15:16:31 LoadBlockIndexDB: last block file = 371
2016-02-08 15:16:31 LoadBlockIndexDB: last block file info: CBlockFileInfo(blocks=8, size=5547601, heights=381890...382910, time=2015-11-03...2015-11-10)
2016-02-08 15:16:31 Checking all blk files are present...
2016-02-08 15:16:31 LoadBlockIndexDB: transaction index disabled
2016-02-08 15:16:32 LoadBlockIndexDB: hashBestChain=0000000000000000062477e780a5d3655c4471a1a94b00f646254003cde6f178 height=382097 date=2015-11-05 01:55:01 progress=0.915093
2016-02-08 15:16:32 init message: Verifying blocks...
2016-02-08 15:16:32 Verifying last 288 blocks at level 3
2016-02-08 15:24:20 No coin database inconsistencies in last 2 blocks (1945 transactions)
2016-02-08 15:24:20  block index          477422ms
2016-02-08 15:24:20 init message: Loading wallet...
2016-02-08 15:24:20 nFileVersion = 110200
2016-02-08 15:24:20 Keys: 101 plaintext, 0 encrypted, 101 w/ metadata, 101 total
2016-02-08 15:24:20  wallet                  163ms
2016-02-08 15:24:20 init message: Activating best chain...
2016-02-08 15:28:14 UpdateTip: new best=00000000000000000d710f3876a7b235aab173042cfe77ff7d0d1972bc279465  height=382098  log2_work=83.560277  tx=91028107  date=2015-11-05 01:57:40 progress=0.915088  cache=42.9MiB(4399tx)
2016-02-08 15:30:37 LevelDB read failure: Corruption: block checksum mismatch
2016-02-08 15:30:37 Corruption: block checksum mismatch
2016-02-08 15:30:49 Error reading from database: Database corrupted

**Comment by laanwj** [comment]
This is a duplicate of #7426. Mostly these corruption issues are caused by faulty hardware.

**Comment by Chaintool** [comment]
Hi Wladimir,

I respectfully disagree with your assertion. A HW error would be evident
regardless of the OS, application SW or system utility. I can search the
HDD file system with my OSX browser, verify it with Discutil and mount it
on a Linux and Windows OS OK.

This is the 3rd failure now on three separate devices, a 65gb high speed
SD card, a 128gb USB2 HDD, a 128gb USB3 HDD.

I wanted to know how to reindex and create a new checksum, is it possible?

Regards,

G*

**Comment by laanwj** [comment]
I have set up tons of nodes, and the only times I had any kind of corruption I tracked it down to faulty hardware. If you tried different storage, then it is possibly a problem with the CPU overheating or memory corrupting.

That doesn't mean there is no software issue, but if it is it is something obscure like interaction between a specific file system and leveldb. These are very hard to diagnose at the least. No one was ever able to, and the pattern of occurences doesn't reveal much either (see issues tagged with "Data corruption").

But reindexing is actually easy: to reindex pass `-reindex` on the command line.
